esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6426,7 +6428,25 @@ bool CWsDfuEx::onDFUFilePublish(IEspContext &context, IEspDFUFilePublishRequest
         Owned<IDistributedFile> oldFile = queryDistributedFileDirectory().createNew(fileDesc);
         oldFile->validate();
 
-        if (!oldFile->renamePhysicalPartFiles(newFileName.str(), nullptr, nullptr, fileDesc->queryDefaultDir()))
+        StringBuffer baseDir;
+        const char *typeStr = nullptr;
+        switch (groupType)
+        {
+            case grp_thor:
+                typeStr = "thor";
+                break;
+            case grp_hthor:
+                typeStr = "hthor";
+                break;
+            case grp_roxie:
+                typeStr = "roxie";
+                break;
+            default:
+                throwUnexpected();
+        }
+        getConfigurationDirectory(directories, "data", typeStr, groupName, baseDir);
```

Comment:
knee jerk reaction was to suggest envGetConfigurationDirectory and avoid the directories member, but I suppose this was done on purpose to avoid the overhead of fetching the env tree every time through?